kernel/kernel-api/src/main/java/io/delta/kernel/internal/CommitRangeFactory.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal;
+
+import static io.delta.kernel.internal.util.Preconditions.checkArgument;
+import static io.delta.kernel.internal.util.Utils.resolvePath;
+
+import io.delta.kernel.Snapshot;
+import io.delta.kernel.engine.Engine;
+import io.delta.kernel.internal.fs.Path;
+import io.delta.kernel.internal.lang.ListUtils;
+import io.delta.kernel.internal.util.FileNames;
+import io.delta.kernel.utils.FileStatus;
+import java.util.List;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class CommitRangeFactory {
+
+  private static final Logger logger = LoggerFactory.getLogger(CommitRangeFactory.class);
+
+  private final CommitRangeBuilderImpl.Context ctx;
+  private final Path tablePath;
+  private final Path logPath;
+
+  CommitRangeFactory(Engine engine, CommitRangeBuilderImpl.Context ctx) {
+    this.ctx = ctx;
+    this.tablePath = new Path(resolvePath(engine, ctx.unresolvedPath));
+    this.logPath = new Path(tablePath, "_delta_log");
+  }
+
+  CommitRangeImpl create(Engine engine) {
+    // Resolve startVersion and try to resolve endVersion if possible
+    long startVersion = resolveStartVersion(engine);
+    Optional<Long> endVersionOpt = tryResolveEndVersion(engine);
```

Comment:
how is having an `endVersion = None` useful/acceptable? shouldn't we throw? shouldn't we at some point draw a line in the sand and say "Okay, now we *have* to know the actual endVersion" ---> and why isn't that point right now and here in the code?

Reply:
I think I need to clarify this better in the code. But basically when endVersion is not provided, we want to use the latest version, which we don't have until we do the listing. But, we want to use the potentially resolved startVersion and endVersion to do that listing if possible. I can try to update the comments/docs here.

Basically, tryResolveEndVersion should only resolve version -> version or timestamp -> version. In the case of default, we need to wait until later.

Reply:
Tried to clarify in comments